cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java

```diff
@@ -273,9 +274,24 @@ private void configureAuthentication(String listenerName, List<String> securityP
                 options.add("oauth.ssl.truststore.type=\"PKCS12\"");
             }
 
-            writer.println(String.format("listener.name.%s.oauthbearer.sasl.server.callback.handler.class=io.strimzi.kafka.oauth.server.JaasServerOauthValidatorCallbackHandler", listenerNameInProperty));
-            writer.println(String.format("listener.name.%s.oauthbearer.sasl.jaas.config=org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required unsecuredLoginStringClaim_sub=\"thePrincipalName\" %s;", listenerNameInProperty, String.join(" ", options)));
-            writer.println(String.format("listener.name.%s.sasl.enabled.mechanisms=OAUTHBEARER", listenerNameInProperty));
+            StringBuilder enabledMechanisms = new StringBuilder();
+            if (oauth.isEnableOauthBearer()) {
+                writer.println(String.format("listener.name.%s.oauthbearer.sasl.server.callback.handler.class=io.strimzi.kafka.oauth.server.JaasServerOauthValidatorCallbackHandler", listenerNameInProperty));
+                writer.println(String.format("listener.name.%s.oauthbearer.sasl.jaas.config=org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required unsecuredLoginStringClaim_sub=\"thePrincipalName\" %s;", listenerNameInProperty, String.join(" ", options)));
+                enabledMechanisms.append("OAUTHBEARER");
+            }
+
+            if (oauth.isEnablePlain()) {
+                addOption(options, ServerPlainConfig.OAUTH_TOKEN_ENDPOINT_URI, oauth.getTokenEndpointUri());
+                writer.println(String.format("listener.name.%s.plain.sasl.server.callback.handler.class=io.strimzi.kafka.oauth.server.plain.JaasServerOauthOverPlainValidatorCallbackHandler", listenerNameInProperty));
+                writer.println(String.format("listener.name.%s.plain.sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required %s;", listenerNameInProperty, String.join(" ", options)));
+                if (enabledMechanisms.length() > 0) {
+                    enabledMechanisms.append(",");
+                }
+                enabledMechanisms.append("PLAIN");
+            }
+
+            writer.println(String.format("listener.name.%s.sasl.enabled.mechanisms=%s", listenerNameInProperty, enabledMechanisms));
```

Comment:
so as far as I understood both mechanisms can be enabled at the same time? How it's going to work in that case?

Reply:
SASL works by client specifying which mechanism it wants to use, and simply attempting to use that mechanism on the listener. If it fails it is free to try with some other mechanism. The broker simply delegates to the requested mechanism if it was configured and enabled, otherwise it responds to the client that the mechanism is not available.

On the server you therefore get multiple enabled SASL mechanisms on the listener:

    listener.name.external-9094.sasl.enabled.mechanisms: OAUTHBEARER, PLAIN

And then each mechanism is properly configured with correct server-side callback handlers: 

    listener.name.external-9094.oauthbearer.sasl.server.callback.handler.class=io.strimzi.kafka.oauth.server.JaasServerOauthValidatorCallbackHandler
    listener.name.external-9094.oauthbearer.sasl.jaas.config=org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required  OPTIONS ...;

    listener.name.external-9094.plain.sasl.server.callback.handler.class: io.strimzi.kafka.oauth.server.plain.JaasServerOauthOverPlainValidatorCallbackHandler
    listener.name.external-9094.plain.sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required OPTIONS ...;